real_estate_module/models/estate_property.py

```diff
@@ -0,0 +1,216 @@
+        comodel_name='res.partner',
+        string="Buyer",
+        help="The buyer who purchased the property"
+    )
+    salesman_id = fields.Many2one(
+        comodel_name='res.users',
+        string="Salesman",
+        default=lambda self: self.env.user,
+        help="The salesperson responsible for the property"
+    )
+    tag_ids = fields.Many2many(
+        comodel_name='estate.property.tags',
+        string="Tags",
+        help="Tags associated with the property (e.g., Luxury, Beachfront)"
+    )
+    offer_ids = fields.One2many(
+        comodel_name='estate.property.offer',
+        inverse_name='property_id',
+        string="Offers",
+        help="Offers made for this property"
+    )
+    # Computed Fields
+    total_area = fields.Float(
+        string="Total Area (sq.m)",
+        compute='_compute_total_area',
+        help="The total area of the property (living area + garden area)"
+    )
+    best_price = fields.Float(
+        string="Best Offer Price",
+        compute='_compute_best_price',
+        help="The highest offer price received for the property"
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string="Company",
+        default=lambda self: self.env.company
+    )
+    # SQL Constraints
+    _sql_constraints = [
+        (
+            'check_expected_price',
+            'CHECK(expected_price > 0)',
+            "The Expected Price must be strictly positive."
+        ),
+        (
+            'check_selling_price',
+            'CHECK(selling_price > 0)',
+            "The Selling Price must be strictly positive."
+        )
+    ]
+    # Compute Total Area
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        """Computes the total property area by adding living area and garden area."""
+        for property in self:
+            property.total_area = property.living_area + property.garden_area
+    # Compute Best Offer Price
+    @api.depends('offer_ids.price')
+    def _compute_best_price(self):
+        """Computes the highest offer price received for the property."""
+        for property in self:
+            property.best_price = max(property.offer_ids.mapped('price'), default=0.0)
+    # Onchange Method for Garden Fields
+    @api.onchange('garden')
+    def _onchange_garden_area_orientation(self):
+        """Automatically set default garden area and orientation when garden is selected."""
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = False
+    # Button Actions
+    def action_cancel_button(self):
+        """Cancel the property sale. Prevent cancellation if the property is already sold."""
+        if self.state == 'sold':
+            raise UserError("Sold properties cannot be cancelled.")
+        self.state = 'cancelled'
+    def action_sold_button(self):
```

Comment:
```suggestion
        self.state = 'cancelled'

    def action_sold_button(self):
```
A space between two methods?